TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -1021,58 +1030,221 @@ private void WriteVariantResults()
+                                    deletionVariants.Add(variantPWSM.Peptide.Protein, new HashSet<SequenceVariation> { variant });
+                                }
+                            }                            
                             else if (culture.CompareInfo.IndexOf(variant.Description.Description, "stop_loss", CompareOptions.IgnoreCase) >= 0)
                             {
-                                stopLossCount++;
-                            }
+
+                                if (stopLossIdentifed == false)
+                                {
+                                    stopLossCount++;
+                                    stopLossIdentifed = true;
+                                }
+                                if (stopLossVariants.ContainsKey(variantPWSM.Peptide.Protein))
+                                {
+                                    stopLossVariants[variantPWSM.Peptide.Protein].Add(variant);
+                                }
+                                else
+                                {
+                                    stopLossVariants.Add(variantPWSM.Peptide.Protein, new HashSet<SequenceVariation> { variant });
+                                }
+
+                            }                          
+                            
+
                         }
-                    }
+                    }               
                 }
             }
+            int SNVmissenseSites = 0;
+            foreach (var entry in SNVmissenseVariants)
+            {
+                SNVmissenseSites += entry.Value.Count;
+            }
+
+            int MNVmissenseSites = 0;
+            foreach (var entry in MNVmissenseVariants)
+            {
+                MNVmissenseSites += entry.Value.Count;
+            }
+
+            int insertionSites = 0;
+            foreach (var entry in insertionVariants)
+            {
+                insertionSites += entry.Value.Count;
+            }
+
+            int deletionSites = 0;
+            foreach (var entry in deletionVariants)
+            {
+                deletionSites += entry.Value.Count;
+            }
 
-            string[] variantResults = new string[15];
+            int frameshiftSites = 0;
+            foreach (var entry in frameshiftVariants)
+            {
+                frameshiftSites += entry.Value.Count;
+            }
+
+            int stopGainSites = 0;
+            foreach (var entry in stopGainVariants)
+            {
+                stopGainSites += entry.Value.Count;
+            }
+
+            int stopLossSites = 0;
+            foreach (var entry in stopLossVariants)
+            {
+                stopLossSites += entry.Value.Count;
+            }
+
+            int totalVariantSites = SNVmissenseSites + MNVmissenseSites+ insertionSites + deletionSites + frameshiftSites + stopGainSites + stopLossSites;          
+
+            
+
+            string[] variantResults = new string[25];
```

Comment:
this is probably better implemented as a `List<string>` instead of a fixed-length array but it's fine for now